include/Contact.php

```diff
@@ -662,4 +664,34 @@ function posts_from_contact($a, $contact_id) {
 
 	return $o;
 }
+
+/**
+ * @brief Returns a formatted location string from the given profile array
+ *
+ * @param array $profile Profile array (Generated from the "profile" table)
+ *
+ * @return string Location string
+ */
+function formatted_location($profile) {
```

Comment:
Mhhh I'm also not really sure. text.php is really bloated. On the other formatted_location doesn't seem to fit totally because it doesn't do anything with the contact and it is more related to profile or identity things. But I don't have a real good  alternative.